crates/chain/src/keychain/txout_index.rs

```diff
@@ -120,6 +112,36 @@ impl<K: Clone + Ord + Debug> Indexer for KeychainTxOutIndex<K> {
 }
 
 impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
+    /// Scans a transaction for relevant outpoints, which are stored and indexed internally.
+    ///
+    /// If the matched script pubkey is part of the lookahead, the last stored index is updated for
+    /// the script pubkey's keychain and the [`super::ChangeSet`] returned will reflect the
+    /// change.
+    ///
+    /// Typically, this method is used in two situations:
+    ///
+    /// 1. After loading transaction data from the disk, you may scan over all the txouts to restore all
+    /// your txouts.
+    /// 2. When getting new data from the chain, you usually scan it before incorporating it into
+    /// your chain state.
+    pub fn scan(&mut self, tx: &bitcoin::Transaction) -> super::ChangeSet<K> {
```

Comment:
Why are we reintroducing `scan` and `scan_txout`? The `Indexer` methods already do the same thing.

Same question for `KeychainTxOutIndex`.

Reply:
I can see how to get rid of `scan` and `scan_txout` in `TxoutIndex` but I don't see how to get rid of it in `SpkTxoutIndex` as `scan_txout` in `TxoutIndex` depends on `scan_txout` in `SpkTxoutIndex` returning an optional index. Do you mind sharing some code sample of how this could work?

Reply:
Thanks for explaining, I see the problem now.

In `KeychainTxOutIndex`, the signature of the methods are exactly the same so we don't need to duplicate them (rm `scam` and `scan_txout`). For `SpkTxOutIndex`, they are different, so we need different methods.